zerver/models.py

```diff
@@ -551,7 +551,7 @@ class UserProfile(AbstractBaseUser, PermissionsMixin):
     bot_owner = models.ForeignKey('self', null=True, on_delete=models.SET_NULL)  # type: Optional[UserProfile]
     long_term_idle = models.BooleanField(default=False, db_index=True)  # type: bool
 
-    USERNAME_FIELD = 'email'
+    USERNAME_FIELD = 'id'
```

Comment:
Hrm. I worry that I'm not quite sure what various bits of Django code might do with this attribute -- [the docs](https://docs.djangoproject.com/en/1.11/topics/auth/customizing/#django.contrib.auth.models.CustomUser.USERNAME_FIELD) leave that pretty open-ended, and various other bits of docs (including elsewhere on that page) mention making use of USERNAME_FIELD.

What breaks if we just don't change this? The docs say it needs to be unique "unless you use a custom authentication backend that can support non-unique usernames" -- and certainly all our auth backends need to support non-unique emails at this point, and I don't think any of them are using `id` as a username in any meaningful sense.

Reply:
Yeah, so I audited this pretty carefully.  The main thing that sucks if we change this is that error message.  The main things that break if we don't change this is that you get a warning whenever you run `manage.py makemigrations` (which can be suppressed using a Django config option) and adding another thing we need to pay attention to when integrating third-party Django modules that might assumed USERNAME_FIELD is unique.  

I think probably we should change this back and suppress the Django warning; you'll be able to get how to do that pretty easily by googling the error message.

Reply:
Cool, sounds good. I think this'll be the one potentially-tricky thing to change in this branch before we merge it, then.